PlayCover/ViewModel/StoreVM.swift

```diff
@@ -6,184 +6,192 @@
+            try data.write(to: plistSource)
+            return true
+        } catch {
+            print("StoreVM: Failed to encode Sources.plist! ", error)
+            return false
         }
     }
 
-    func moveSourceDown(_ selected: inout Set<UUID>) {
-        let selectedData = self.sources.filter({ selected.contains($0.id) })
+    //
+    @discardableResult private func decode() -> Bool {
+        do {
+            let data = try Data(contentsOf: plistSource)
+            sourcesList = try PropertyListDecoder().decode([SourceData].self, from: data)
+            return true
+        } catch {
+            print("StoreVM: Failed to decode Sources.plist! ", error)
+            return false
+        }
+    }
 
-        if let first = selectedData.first {
-            if var index = self.sources.firstIndex(of: first) {
-                index += 1
-                self.sources.removeAll(where: { selected.contains($0.id) })
-                if index > self.sources.endIndex {
-                    index = self.sources.endIndex
+    //
+    private func getSourceData(sourceLink: String) async -> (SourceJSON?, SourceValidation) {
+        guard let url = URL(string: sourceLink) else { return (nil, .badurl) }
+        do {
+            let (data, response) = try await URLSession.shared.data(
+                for: URLRequest(url: url, cachePolicy: .reloadIgnoringLocalAndRemoteCacheData)
+            )
+            if !url.isFileURL {
+                guard (response as? HTTPURLResponse)?.statusCode == 200 else { return (nil, .badurl) }
+            }
+            let jsonResult: SourceJSON = try JSONDecoder().decode(SourceJSON.self, from: data)
+            debugPrint("SourceJSON from \(url) Fetched")
+            return (jsonResult, .valid)
+        } catch {
+            do {
+                let (data, response) = try await URLSession.shared.data(
+                    for: URLRequest(url: url, cachePolicy: .reloadIgnoringLocalAndRemoteCacheData)
+                )
+                if !url.isFileURL {
+                    guard (response as? HTTPURLResponse)?.statusCode == 200 else { return (nil, .badurl) }
                 }
-                self.sources.insert(contentsOf: selectedData, at: index)
+                let jsonResult: [SourceAppsData] = try JSONDecoder().decode([SourceAppsData].self, from: data)
+                let newJson = SourceJSON(name: url.isFileURL ? "localhost" : url.host ?? url.absoluteString,
+                                         logo: "NoLogo",
+                                         data: jsonResult)
+                return (newJson, .valid)
+            } catch {
+                debugPrint("Error decoding data from URL: \(url): \(error)")
+                return (nil, .badjson)
             }
         }
     }
+}
 
-    func appendSourceData(_ data: SourceData) {
-        self.sources.append(data)
-        self.resolveSources()
-    }
+// Source Data Structure
```

Comment:
These structs would be better on a different file

Reply:
we have no file for data structs, it is somehow redundant to create one just for source structs; isn't it?